test/functional/terminal/highlight_spec.lua

```diff
@@ -134,7 +134,7 @@ describe('terminal window highlighting with custom palette', function()
     })
     screen:attach(true)
     nvim('set_var', 'terminal_color_3', '#123456')
-    execute('enew | call termopen(["'..nvim_dir..'/tty-test"]) | startinsert')
+    execute('enew | let $NVIM_TUI_ENABLE_TRUE_COLOR=1 | call termopen(["'..nvim_dir..'/tty-test"]) | startinsert')
```

Comment:
Ah, ok. I guess this will be solved if I use ui_rgb_attached() as you suggested below.